roles/common/tasks/networking.yml

```diff
@@ -13,6 +13,16 @@
   with_items: network_interfaces_d
   tags: ['prereboot']
 
+- name: get uname -n
+  command: uname -n
+  register: uname
+  tags: etc_hosts
+
+- name: create 'uname -n' as fact 'uname_host'
+  set_fact:
+    uname_host: "{{ uname.stdout }}"
+  tags: etc_hosts
```

Comment:
Ah, so platform.node() on Linux does a uname -n equivalent. So ansible_nodename should always match uname -n.